.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+name: CI
+on:
+  push:
+    branches: [develop]
+  pull_request:
+
+env:
+  CXXFLAGS: "-fprofile-arcs -ftest-coverage"
+
+jobs:
+  test:
+    name: "Python ${{ matrix.python-version }} / ${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["2.7", "3.8"]
+        os: [ubuntu-latest]
+        compiler: [gcc]
+    env:
+      CC: ${{ matrix.compiler }}
+      PY2: ${{ startsWith(matrix.python-version, 2) && 'on' || 'off' }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install -qq libboost-all-dev swig libhdf5-serial-dev libeigen3-dev \
+            python-numpy python3-biopython python3-numpy cmake libcgal-dev libcgal-qt5-dev \
+            libfftw3-dev libopencv-dev libgsl0-dev python2-dev python3-dev libann-dev \
+            libprotobuf-dev protobuf-compiler python-protobuf python3-protobuf \
+            libopenmpi-dev
+      - name: Set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: 'x64'
+      - name: Set up git
+        run: ./setup_git.py
+      - name: Build IMP
+        run: |
+          mkdir build
+          cd build
+          cmake .. -DUSE_PYTHON2=${{ env.PY2 }} -DCMAKE_CXX_FLAGS="${{ env.CXXFLAGS }}"
+          make -k -j 2
+      - name: Run tests
+        run: ./setup_environment.sh ../test/nosetests --with-coverage --cover-branches -e 'probabilistic' ../test/test_*.py ../test/medium_test_*.py test/*/*.py > /dev/null
```

Comment:
`-e 'probabilistic'` says to ignore all tests that match `probabilistic`, right? Why is this included?

Reply:
There used to be a utility method in `IMP.test` called `probabilistic_test`. nose would run it as a test, but it's not a test and so would always fail; see #996. That workaround shouldn't be needed any more though since I renamed the method  to `probabilistic_check`.

The other workaround is to symlink the nosetests binary into the test directory and run it there, otherwise methods like `get_input_file_name` don't work. But I think I fixed that as part of #996 too.